### ISSUE TITLE
[Project Addition]: Web Application for Deaf Community

### ISSUE THREAD

**Issue by vishuhere** [abhisheks008/DL-Simplified#564]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** :  Web Application for Deaf Community</br>
:red_circle: **Aim**: Assisting individuals with hearing impairments to facilitate communication with the Deaf community</br>
:red_circle: **Dataset** : Gesture Recognition Datasets </br>
:red_circle: **Approach**: Finding the movements of hands through pre-trained models like media pipe and VITpose then taking the model with the most precise output and then assigning keywords to every movement of actions.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : VIshwajeet Gehlot
- GitHub Profile Link : https://github.com/vishuhere
- Email ID : vigehlot3@gmail.com
- Approach for this Project : Finding the movements of hands through pre-trained models like media pipe and VITpose then taking the model with the most precise output and then assigning keywords to every movement of actions.
- What is your participant role? (Mention the Open Source program): GSSOC 2024 Contributor

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by abhisheks008** [comment]
Can you share the dataset URL? Also what are the algorithms you are planning to implement here, can you specify them?
@vishuhere

**Comment by vishuhere** [comment]
Custom Dataset: I will use my own specific application, I will collect my own dataset by recording hand gestures using a webcam as I have a specific set of gestures relevant to my use case.

CNN Dataset will be used.

For Dataset url: It's not created yet.I will implement it from the scratch as I had already worked on this thing you can go through my github for more info. Also feel free to reach me anytime for such query I'm happy to clear your doubts.

PS: While assigning the task kindly put GSSOC with level3 on it.Thanks.

**Comment by abhisheks008** [comment]
What are the algorithms you are planning to implement here, can you specify them? And for the level tag, initially all the deep learning issues are being given as `level2`, after judging the pull request and your contribution it can be upgraded to `level3`.
@vishuhere

**Comment by vishuhere** [comment]
Algorithmic steps involved in developing the model through hand gesture recognition:

1) Data Collection: I will Gather a dataset of hand gesture images or videos along with their corresponding labels.

2) Data Preprocessing: I will Preprocess the dataset to ensure uniformity and compatibility, including tasks like resizing images, normalization, and augmentation to increase dataset variability.

3) Feature Extraction: Extract relevant features from the preprocessed images or videos. This may involve Convolutional Neural Networks (CNNs) technique for automatic feature learning.

4) Model Selection: This part will be handled by Conventional Neural Networks (CNN).

5) Model Training: Training the selected model using the Custom dataset.

6) Model Evaluation: Evaluate the trained model's performance on a separate validation or test dataset to assess its accuracy and precision.

7) Deployment: Integrating the model into the website, or embedded system for real-time recognition.

**Comment by abhisheks008** [comment]
Looks good to me! Issue assigned to you @vishuhere
